src/UI/CompletionsView.re

```diff
@@ -68,192 +63,186 @@ let completionKindToIcon: option(Ext.CompletionItemKind.t) => int =
-                  backgroundColor=completionColor
-                  color=bgColor
-                  margin=4
-                  // Not sure why, but specifying a font size fails to render the icon!
-                  // Might be a bug with Revery font loading / re - rendering in this case?
-                />
-              </View>
-              <View style=Style.[flexGrow(1), margin(4)]>
-                <HighlightText
-                  highlights={curr.highlight}
-                  style=normalStyle
-                  highlightStyle
-                  text=message
-                />
-              </View>
-            </View>;
-          let newDiags = [newElem, ...prevDiags];
-          (newWidth, newDiags);
-        },
-        (maxCompletionWidth, []),
-        completions.filteredCompletions,
-      );
-
-    let totalWidth = _maxWidth + padding * 2;
-
-    let detailElem =
-      completions
-      |> Model.Completions.getBestCompletion
-      |> Option.bind(
-           (filteredCompletion: Model.Completions.filteredCompletion) =>
-           filteredCompletion.item.detail
-         )
-      |> Option.bind(text =>
-           if (String.length(text) > 0) {
-             Some(
-               <View style={detailStyle(totalWidth)}>
-                 <Text style=detailTextStyle text />
-               </View>,
-             );
-           } else {
-             None;
-           }
-         )
-      |> Option.value(~default=React.empty);
-
-    let completionItems = List.rev(completionItems) |> React.listToElement;
-
-    <View style=outerPositionStyle>
-      <Opacity opacity>
-        <View style={innerPositionStyle(totalWidth)}> completionItems </View>
-        detailElem
-      </Opacity>
-    </View>;
-  };
+  let {theme, tokenTheme, editorFont, completions, _}: State.t = state;
+  let items = completions.filtered;
+
+  let maxWidth =
+    items
+    |> Array.fold_left(
+         (maxWidth, this: Filter.result(CompletionItem.t)) => {
+           let textWidth =
+             EditorFont.measure(~text=this.item.label, editorFont);
+           let thisWidth = int_of_float(textWidth +. 0.5) + Constants.padding;
+           max(maxWidth, thisWidth);
+         },
+         Constants.maxCompletionWidth,
+       );
+
+  let width = maxWidth + Constants.padding * 2;
+  let height =
+    min(Constants.maxHeight, Array.length(items) * Constants.itemHeight);
+
+  let detail =
+    switch (completions.focused) {
+    | Some(index) =>
+      let focused: Filter.result(CompletionItem.t) = items[index];
+      switch (focused.item.detail) {
+      | Some(text) =>
+        <detailView text width lineHeight theme tokenTheme editorFont />
+      | None => React.empty
+      };
+    | None => React.empty
+    };
+
+  <View style={Styles.outerPosition(~x, ~y)}>
+    <Opacity opacity=Constants.opacity>
+      <View
+        style={Styles.innerPosition(~height, ~width, ~lineHeight, ~theme)}>
+        <FlatList
```

Comment:
Having the scroll bar and being able to move through completions is awesome - so nice to have that now!
![image](https://user-images.githubusercontent.com/13532591/71940181-3b55e200-316a-11ea-99ed-ca84dc0a6fbd.png)

🎉 🎉